lib/src/component/resize/resize_sensor.dart

```diff
@@ -0,0 +1,317 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Thanks!
+/// https://github.com/marcj/css-element-queries/blob/master/src/ResizeSensor.js
+library resize_sensor;
+
+import 'dart:html';
+
+import 'package:meta/meta.dart';
+import 'package:platform_detect/platform_detect.dart';
+
+import 'package:over_react/over_react.dart';
+import 'package:over_react/src/component/resize/resize_props_mixin.dart';
+
+/// A wrapper component that detects when its parent is resized.
+///
+/// This component _must_ be put in a relative or absolutely positioned
+/// container.
+///
+///     (ResizeSensor()..onResize = () => print('resized'))(children)
+///
+/// See: <https://docs.workiva.org/web_skin_dart/latest/components/#resize-sensor>.
+@Factory()
+UiFactory<ResizeSensorProps> ResizeSensor;
```

Comment:
IMO its a little confusing that this factory / props class / component class are named identically to the one we export.

Reply:
I agree, do you have a suggestion?